core/src/services/gcs/core.rs

```diff
@@ -117,6 +119,12 @@ impl GcsCore {
         }
         let cred = self.load_token().await?;
 
+        if !self.token.is_empty() {
```

Comment:
If a token has been set, we can return it directly. This can happen before `allow_anonymous` and `load_token`.

By the way, did I overlook part of the previous PR? This `if self.allow_anonymous {}` check doesn't seem correct.

Reply:
> This if self.allow_anonymous {} check doesn't seem correct.

I was wondering the same thing after I saw it here too! I can raise a separate PR to address this quickly.

If I understand rightly, this can be entirely removed? It already exists within the `load_credential` part, i.e. "allow anonymous access" not "skipping signing entirely".

Reply:
I'm concerned that we still need to adjust the behavior here a bit, such as skipping the sign if the token is empty. I also find that `load_token` isn't handling things correctly.